src/ipc/ipc4/handler.c

```diff
@@ -1003,15 +1003,49 @@ void ipc_boot_complete_msg(struct ipc_cmd_hdr *header, uint32_t data)
 }
 
 #ifdef CONFIG_LOG_BACKEND_ADSP_MTRACE
+
+static bool is_notification_queued(void)
+{
+	struct ipc *ipc = ipc_get();
+	struct list_item *slist;
+	struct ipc_msg *msg;
+	k_spinlock_key_t key;
+	bool queued = false;
+
+	key = k_spin_lock(&ipc->lock);
+	if (list_is_empty(&ipc->msg_list))
+		goto out;
+
+	list_for_item(slist, &ipc->msg_list) {
+		msg = container_of(slist, struct ipc_msg, list);
+
+		if (msg == &msg_notify) {
```

Comment:
are you over writing the list here ?

Reply:
@lgirdwood intention is the reverse, not to touch ipc->msg_list if the statically allocated "msg_notify" is already on the list.

Reply:
Should there be similar test for the statically allocated msg_reply ?

Reply:
It is highly unlikely that it ever catches anything (it must not), but what if it does?

Reply:
Let me add that to local tree first and see if I get any hits. This should not happen as ipc_cmd() runs the handle+wait+respond synchronously and the IPC EDF thread is pinned on core0.